api/handlers_e2e_test.go

```diff
@@ -1669,6 +1637,151 @@ func TestGetBlocksTransactionsLimit(t *testing.T) {
 	}
 }
 
+func TestGetBlockWithCompression(t *testing.T) {
+	db, shutdownFunc, proc, l := setupIdb(t, test.MakeGenesisV2())
+	defer shutdownFunc()
+	defer l.Close()
+
+	///////////
+	// Given // A block containing 20 transactions at round 1
+	//       //
+	///////////
+
+	const numbOfTxns = 20
+	var txns []transactions.SignedTxnWithAD
+	for j := 0; j < numbOfTxns; j++ {
+		txns = append(txns, test.MakePaymentTxn(1, 100, 0, 0, 0, 0, test.AccountA, test.AccountB, basics.Address{}, basics.Address{}))
+	}
+	ptxns := make([]*transactions.SignedTxnWithAD, numbOfTxns)
+	for k := range txns {
+		ptxns[k] = &txns[k]
+	}
+	block, err := test.MakeBlockForTxns(test.MakeGenesisBlock().BlockHeader, ptxns...)
+	block.BlockHeader.Round = basics.Round(1)
+	require.NoError(t, err)
+
+	err = proc(&rpcs.EncodedBlockCert{Block: block})
+	require.NoError(t, err)
+
+	//////////
+	// When // We look up a block using a ServerImplementation with a compression flag on/off
+	//////////
+
+	serverCtx, serverCancel := context.WithCancel(context.Background())
+	defer serverCancel()
+	opts := defaultOpts
+	listenAddr := "localhost:8889"
+	go Serve(serverCtx, listenAddr, db, nil, logrus.New(), opts)
+
+	waitForServer(t, listenAddr)
+
+	getBlockFunc := func(t *testing.T, headerOnly bool, useCompression bool) *generated.BlockResponse {
+		path := "/v2/blocks/1"
```

Comment:
If we make this a parameter, we can generalize this across a bunch of different APIs and test to make sure compression works in general and not just for blocks.

Reply:
Since we're just testing the blocks endpoint for now, we can pull this into a separate function for other APIs later.